### PR TITLE
HPCC-22210 Fix some uninitialized members in join helpers

### PR DESCRIPTION
**Pull request by jakesmith**
Signed-off-by: Jake Smith <jake.smith@lexisnexisrisk.com>

## Type of change:
- [x] This change is a bug fix (non-breaking change which fixes an issue).
- [ ] This change is a new feature (non-breaking change which adds functionality).
- [ ] This change improves the code (refactor or other change that does not change the functionality)
- [ ] This change fixes warnings (the fix does not alter the functionality or the generated code)
- [ ] This change is a breaking change (fix or feature that will cause existing behavior to change).
- [ ] This change alters the query API (existing queries will have to be recompiled)

## Checklist:
- [x] My code follows the code style of this project.
  - [ ] My code does not create any new warnings from compiler, build system, or lint.
- [x] The commit message is properly formatted and free of typos.
  - [ ] The commit message title makes sense in a changelog, by itself.
  - [ ] The commit is signed.
- [ ] My change requires a change to the documentation.
  - [ ] I have updated the documentation accordingly, or...
  - [ ] I have created a JIRA ticket to update the documentation.
  - [ ] Any new interfaces or exported functions are appropriately commented.
- [x] I have read the CONTRIBUTORS document.
- [x] The change has been fully tested:
  - [ ] I have added tests to cover my changes.
  - [ ] All new and existing tests passed.
  - [ ] I have checked that this change does not introduce memory leaks.
  - [ ] I have used Valgrind or similar tools to check for potential issues.
- [x] I have given due consideration to all of the following potential concerns:
  - [ ] Scalability
  - [ ] Performance
  - [ ] Security
  - [ ] Thread-safety
  - [ ] Premature optimization
  - [ ] Existing deployed queries will not be broken
  - [ ] This change fixes the problem, not just the symptom
  - [ ] The target branch of this pull request is appropriate for such a change.
- [x] There are no similar instances of the same problem that should be addressed
  - [ ] I have addressed them here
  - [ ] I have raised JIRA issues to address them separately
- [ ] This is a user interface / front-end modification
  - [ ] I have tested my changes in multiple modern browsers
  - [ ] The component(s) render as expected

## Smoketest:
- [ ] Send notifications about my Pull Request position in Smoketest queue.
- [ ] Test my draft Pull Request.

### CODE REVIEW

**Comment by jakesmith** [comment]
@mckellyln - please check.

File: thorlcr/activities/msort/thsortu.cpp [hunk 1]
```diff
@@ -294,44 +292,44 @@ class CJoinHelper : implements IJoinHelper, public CSimpleInterface
...
-    bool eofL;
-    bool eofR;
+    unsigned denormCount = 0;
+    unsigned joinCounter = 0;
+    size32_t outSz = 0;
+    unsigned rightidx = 0;
+    enum { JScompare, JSmatch, JSrightgrouponly, JSonfail } state = JScompare;
+    bool eofL = false;
+    bool eofR = false;
     IHThorJoinArg *helper;
```
Comment (mckellyln) [inline thorlcr/activities/msort/thsortu.cpp]: ok, why not ?

File: thorlcr/activities/msort/thsortu.cpp [hunk 2]
```diff
@@ -474,6 +468,13 @@ class CJoinHelper : implements IJoinHelper, public CSimpleInterface
         outputmetaL = _outputmeta;
         if (TAKdenormalize == kind || TAKhashdenormalize == kind || TAKlookupdenormalize == kind || TAKsmartdenormalize == kind)
             denormTmp.setAllocator(allocator).ensureRow();
+        denormCount = 0;
+        joinCounter = 0;
+        outSz = 0;
+        rightidx = 0;
+        leftmatched = false;
+        lhsProgressCount = 0;
+        rhsProgressCount = 0;
```
Comment (mckellyln) [inline thorlcr/activities/msort/thsortu.cpp]: ok, why not all the same ones as before ?

**Comment by mckellyln** [comment]
@jakesmith looks fine, but two minor questions.

File: thorlcr/activities/msort/thsortu.cpp (hunk 1 above)
Comment (jakesmith) [reply thorlcr/activities/msort/thsortu.cpp]: it's always initialized by the ctor, but still I prefer all member declarations to be initialized.
I'll fix.

File: thorlcr/activities/msort/thsortu.cpp (hunk 2 above)
Comment (jakesmith) [reply thorlcr/activities/msort/thsortu.cpp]: they should be, which ones look different?

**Comment by jakesmith** [comment]
@mckellyln - fixed the helper member initialization issue.

File: thorlcr/activities/msort/thsortu.cpp (hunk 1 above)
Comment (mckellyln) [reply thorlcr/activities/msort/thsortu.cpp]: As is kind, which is only reasone why I asked - it seemed to be the one left out.

File: thorlcr/activities/msort/thsortu.cpp (hunk 2 above)
Comment (mckellyln) [reply thorlcr/activities/msort/thsortu.cpp]: Maybe I missed it, is this one there ?
```
        limitedCompareR = NULL;
```

Comment (mckellyln) [reply thorlcr/activities/msort/thsortu.cpp]: ok, maybe it is unused and removed ?

Comment (jakesmith) [reply thorlcr/activities/msort/thsortu.cpp]: ah that one - yes removed - it was not used.

**Review by mckellyln — APPROVED** [review]

**Comment by mckellyln** [comment]
@jakesmith looks good to merge.

**Comment by jakesmith** [comment]
@richardkchapman - please merge.

**Pull request merged**